### PR TITLE
Search Page With Filters:

### PR DESCRIPTION
**Pull request by caneppelevitor**
- Manage filter query parameter and adapted functions to make it possible search with and without filter parameters
- Adapted Sentence findAll to be able to search by filter without necessary a searchText
- Create a Advanced Search Component to manage filters on the search
- Made topic tags clickable redirecting user to search page with the tag as the main filter

### CODE REVIEW

File: src/api/topicsApi.ts [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
 import { message } from "antd";
+import { ActionTypes } from "../store/types";
 
 const request = axios.create({
     withCredentials: true,
     baseURL: `/api/topics`,
 });
 
-const getTopics = (topicName, t) => {
+const getTopics = (dispatch, topicName, t) => {
```
Comment (thesocialdev) [inline src/api/topicsApi.ts:10]: suggestions: if you're going to pass optional parameters, pass an object as parameter and do the following:
```
const getTopics = ({ dispatch, topicName, t }) => {
```

Comment (thesocialdev) [reply src/api/topicsApi.ts:10]: Also, please create a type interface for this function

File: server/search/search.controller.ts [hunk 2]
```diff
@@ -125,7 +135,12 @@ export class SearchController {
                     req.language,
                     null
                 ),
-                this.sentenceService.findAll(searchText, parsedPageSize, null),
+                this.sentenceService.findAll(
+                    searchText,
+                    parsedPageSize,
+                    null,
```
Comment (thesocialdev) [inline server/search/search.controller.ts:141]: suggestion: make it optional in the function statement and just omit the parameter when calling the function

File: server/claim/types/sentence/sentence.service.ts [hunk 3]
```diff
@@ -123,7 +111,45 @@ export class SentenceService {
...
+                        path: "topics",
+                        fuzzy: {
+                            maxEdits: 2,
+                        },
+                    },
+                },
+            });
+        }
+
+        if (filter) {
```
Comment (thesocialdev) [inline server/claim/types/sentence/sentence.service.ts:144]: question: what happens if searchText is provided without filter? Also, does the order matter?

File: server/claim/types/sentence/sentence.service.ts [hunk 4]
```diff
@@ -123,7 +111,45 @@ export class SentenceService {
                     },
                 },
             },
-        ]);
+        ];
+
+        if (searchText) {
+            pipeline.splice(0, 0, {
```
Comment (thesocialdev) [inline server/claim/types/sentence/sentence.service.ts:117]: suggestion: The splice logic is a bit confusing to read, I suggest that we just push them in the order that matters. It seems that you're using splice just to add the pipelines in the right order, maybe we should just push them in the right order starting from an empty array, what do you think?

File: server/search/search.controller.ts [hunk 5]
```diff
@@ -36,35 +36,44 @@ export class SearchController {
...
-                        searchText,
-                        processedPageSize,
-                        skipedDocuments
-                    ),
-                    this.claimRevisionService.findAll(
-                        searchText,
-                        processedPageSize,
-                        skipedDocuments
-                    ),
+                    searchText
```
Comment (thesocialdev) [inline server/search/search.controller.ts:52]: suggestion: convert each of these promises into a separate function and use async/await

File: src/components/ClaimReview/TopicInput.tsx [hunk 6]
```diff
@@ -27,7 +27,7 @@ const TopicInput = ({ contentModel, data_hash, topics }) => {
     const [duplicatedErrorMessage, setDuplicatedErrorMessage] = useState("");
 
     const fetchTopicList = async (topic) => {
-        const topicSearchResults = await topicApi.getTopics(topic, t);
+        const topicSearchResults = await topicApi.getTopics(null, topic, t);
```
Comment (thesocialdev) [inline src/components/ClaimReview/TopicInput.tsx:30]: See my comment in the `topicApi` file

File: src/components/Search/AdvancedSearch.tsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+            id="tags-outlined"
+            defaultValue={
+                Array.isArray(defaltValue) ? defaltValue : [defaltValue]
+            }
+            options={options.map((option) => option.name)}
+            onChange={(event, newValue) => handleFilter(newValue)}
+            renderInput={(params) => (
+                <TextField
+                    {...params}
+                    style={{ backgroundColor: "#f8f9fa" }}
```
Comment (thesocialdev) [inline src/components/Search/AdvancedSearch.tsx:23]: issue: get the color from the colors utils/library object

File: src/components/Search/SearchPageView.tsx [hunk 8]
```diff
@@ -133,21 +168,38 @@ function SearchPageView({ searchText }) {
...
-                                <SearchOutlined />
+                                <SearchIcon />
                             </Button>
                         </div>
+                        <AdvancedSearch
+                            defaltValue={filtersUsed}
+                            onSearch={fetchTopicList}
+                            options={autoCompleteTopicsResults}
+                            handleFilter={handleFilter}
+                            t={t}
```
Comment (thesocialdev) [inline src/components/Search/SearchPageView.tsx:201]: issue: do not pass `t` as a prop, use the hook inside `AdvancedSearch` instead.

**Review by thesocialdev — CHANGES_REQUESTED** [review]

**Review by thesocialdev — CHANGES_REQUESTED** [review]
Looks great! I have a few comments to improve the code readability and a couple of issues.

File: server/claim/types/sentence/sentence.service.ts (hunk 3 above)
Comment (caneppelevitor) [reply server/claim/types/sentence/sentence.service.ts:144]: Is gonna bring a common search with all results based on the searchText. And about the order of what specific, the splice on the pipe line or about the searchText and filter ?

File: server/claim/types/sentence/sentence.service.ts (hunk 4 above)
Comment (caneppelevitor) [reply server/claim/types/sentence/sentence.service.ts:117]: I didn't have this ideia, thanks for the suggestion.

**Review by thesocialdev — APPROVED** [review]

**Pull request merged**